### PR TITLE
init binom never powers

### PR DESCRIPTION
**Pull request by C-h-r-i-s-x**
updating 3. proof - filled in sorrys in first step

### CODE REVIEW

File: src/chapters/03_Binomial_coefficients_are_(almost)_never_powers.lean [hunk 1]
```diff
@@ -26,38 +28,139 @@ open nat
...
       { split,
-        { sorry, },
+        -- prove p^l ≤ n
+        { -- p^l ∣ choose n k
+          have h_p_div2 : p ∣ m^l := by
+            { rw ← H, 
+            exact h_p_div, }, 
+          have h_pl_div : p^l ∣ m^l := by
+            { sorry, }, 
+          have h_pl_div2 : p^l  ∣ choose n k := by
```
Comment (mo271) [inline src/chapters/03_Binomial_coefficients_are_(almost)_never_powers.lean:108]:
```suggestion
          have h_pl_div2 : p^l ∣ choose n k := by
```

File: src/chapters/03_Binomial_coefficients_are_(almost)_never_powers.lean [hunk 2]
```diff
@@ -26,38 +28,139 @@ open nat
...
+          { exact nat.pow_lt_pow_of_lt_left (h_pk)(gt_of_ge_of_gt h_l two_pos), },
+          -- prove k² ≤ k^l
+          { exact nat.pow_le_pow_of_le_right (pos_of_gt h_k) h_l,  }, 
+        }, 
+      }, 
+    },
+      
+    -- STEP (2) : Breakdown of numerator factros n-j = a_j m_j^l whereas a_j pairwise distinct
+    have h₂ : ∀ (n k p : ℕ) (h_kleqn : k ≤ n) (h_klp : k < p), 
+    choose n k = ∏ i  in Icc (n-k+1) n , i * 1/k.factorial := by
```
Comment (mo271) [inline src/chapters/03_Binomial_coefficients_are_(almost)_never_powers.lean:144]:
```suggestion
    choose n k = ∏ i in Icc (n - k + 1) n , i * 1/k.factorial := by
```

File: src/chapters/03_Binomial_coefficients_are_(almost)_never_powers.lean [hunk 3]
```diff
@@ -88,4 +191,4 @@ begin
       zify at h_n,
       exact le_sub.mp h_n, },
     exact h_wlog k' h_k_l h_n_l this, }
-end
+end
```
Comment (mo271) [inline src/chapters/03_Binomial_coefficients_are_(almost)_never_powers.lean:194]: I would leave the newline at the end of the file...

**Review by mo271 — APPROVED** [review]

**Comment by mo271** [comment]
Looks great, thanks!

File: src/chapters/03_Binomial_coefficients_are_(almost)_never_powers.lean [hunk 4]
```diff
@@ -26,38 +28,139 @@ open nat
...
+  simp,
+  apply mul_comm,
+end
+
 /-
+### Erdo's Theorem
 Using ℕ instead of ℤ here, because of the definition of `choose` and because of the inequalities.
 -/
-theorem binomials_coefficients_never_powers (k l m n : ℕ) (h_l : l ≥ 2) (h_k : 4 ≤ k)
+theorem binomials_coefficients_never_powers (k l m n : ℕ) (h_l : 2 ≤ l) (h_k : 4 ≤ k)
```
[the remark this answers is not in the document]
Comment (C-h-r-i-s-x) [reply src/chapters/03_Binomial_coefficients_are_(almost)_never_powers.lean:79]: I will fix that later as it destroys the proof.

**Pull request merged**